### ISSUE TITLE
Confusion on understanding some parts of the code

### ISSUE THREAD

**Issue by easquelcannotcode** [blue-blue272/VideoReID-TCLNet#3]
Thanks for the outstanding work you have done as well as sharing your code generously.  But sorry I am a little confused with some parts of the source code. I would be appreciated if you could explain them.

(1) In the "TSB.py", I noticed that you used a query-key-value (self-attention) strategy in enhancing representations, but what is the use of "mask" in the following code? And what is the use of the hyperparameter 1e8? And how do you choose it?
```
# mask the self-enhance
mask = torch.eye(t).type(x.dtype) 
if self.use_gpu: mask = mask.cuda()
mask = mask.view(1, t, t, 1)

f = (f - mask * 1e8).view(b, t, t * h * w)
f = F.softmax(f, dim=-1)
```
(2) Same confusion occurs in similar code of "TSE.py"
```
def erase_feature(self, x, masks, soft_masks):
        """erasing the x with the masks, the softmasks for gradient back-propagation. 
        """
        b, c, h, w = x.size()
        soft_masks = soft_masks - (1 - masks) * 1e8 # This is confusing
        soft_masks = F.softmax(soft_masks.view(b, h * w) , 1)

        inputs = x * masks.unsqueeze(1)
        res = torch.bmm(x.view(b, c, h * w), soft_masks.unsqueeze(-1)) 
        outputs = inputs + self.conv_erase(res.unsqueeze(-1)) # [b, c, h, w]
        return outputs
```

**Comment by blue-blue272** [comment]
@easquelcannotcode  with the use of the hyperparameter 1e8, the value can be about negative infinity. After softmax function, these values are about 0. For TSB, it used to mask the self-connection (set the self-connection to 0), and for TSE, it used to erase the salient regions (set these values to 0).

**Comment by easquelcannotcode** [comment]
> @easquelcannotcode with the use of the hyperparameter 1e8, the value can be about negative infinity. After softmax function, these values are about 0. For TSB, it used to mask the self-connection (set the self-connection to 0), and for TSE, it used to erase the salient regions (set these values to 0).

Thans a lot! BTW, I am the freshman of UCAS graduate school of year 2020. So 非常感谢学长！
